include/fs/fcb.h

```diff
@@ -21,50 +21,106 @@ extern "C" {
 
 #include <kernel.h>
 
-#define FCB_MAX_LEN	(CHAR_MAX | CHAR_MAX << 7) /* Max length of element */
+/**
+ * @defgroup fcb Flash Circular Buffer (FCB)
+ * @ingroup file_system_storage
+ * @{
+ * @}
+ */
 
-/*
+/**
+ * @defgroup fcb_data_structures Flash Circular Buffer Data Structures
+ * @ingroup fcb
+ * @{
+ */
+
+#define FCB_MAX_LEN	(CHAR_MAX | CHAR_MAX << 7) /**< Max length of element */
+
+/**
+ * @brief FCB entry info structure. This data structure describes the element
+ * location in the flash.
+ *
+ * You would use it figure out what parameters to pass to flash_area_read()
+ * to read element contents. Or to flash_area_write() when adding a new element.
  * Entry location is pointer to area (within fcb->f_sectors), and offset
  * within that area.
  */
 struct fcb_entry {
-	struct flash_sector *fe_sector;	/* ptr to sector */
-					/* within fcb->f_sectors */
-	u32_t fe_elem_off;		/* start of entry */
-	u32_t fe_data_off;		/* start of data */
-	u16_t fe_data_len;		/* size of data area */
+	struct flash_sector *fe_sector;
+	/**< Pointer to info about sector where data are placed */
+
+	u32_t fe_elem_off;
+	/**< Offset from the start of the sector to beginning of element. */
+
+	u32_t fe_data_off;
+	/**< Offset from the start of the sector to beginning of element. */
```

Comment:
the beginning. Also, perhaps better to use either "start" or "beginning", but consistently in these two sentences